syntax/gleam.vim

```diff
@@ -8,32 +8,65 @@ let b:current_syntax = "gleam"
 
 " Keywords
 syntax keyword gleamKeyword
-  \ module import pub panic use
-  \ type let as if else todo const
-  \ case assert tuple try opaque
+  \ import pub panic use type let as if else todo const case assert try opaque
 highlight link gleamKeyword Keyword
 
 " Function definition
 syntax keyword gleamDef fn nextgroup=gleamFunctionDef skipwhite skipempty
 highlight link gleamDef Keyword
 
-syntax match gleamFunctionDef "[a-z_-][0-9a-z_-]*" contained skipwhite skipnl
+syntax match gleamFunctionDef "[a-z][a-z0-9_]*\ze\s*(" skipwhite skipnl
+highlight link gleamFunctionDef Function
 highlight link gleamFunctionDef Function
 
-" Int
-syntax match gleamInt '\<\(0x[a-fA-F0-9_]\+\|[0-9][0-9_]*\)\>'
+" Number
+"" Int
+syntax match gleamInt '\<\d[0-9_]*\>'
 highlight link gleamInt Number
 
-" Float
-syntax match gleamFloat '\<[0-9][0-9_]*\.[0-9_]*\>'
+"" Binary
+syntax match gleamBinary '\<0[bB][01]+\>'
+highlight link gleamBinary Number
+
+"" Octet
+syntax match gleamOctet '\<0[oO][0-7]+\>'
+highlight link gleamOctet Number
+
+"" Hexadecimal
+syntax match gleamHexa '\<0[xX]\x+\>'
+highlight link gleamHexa Number
+
+"" Float
+syntax match gleamFloat '\d\(_\=\d\)*\.\(\d\(_\=\d\)*\)\=\(e-\=\d\)\=[0-9_]*'
 highlight link gleamFloat Float
 
 " Operators
-syntax match gleamOperator "\([-!#$%`&\*\+./<=>@\\^|~:]\|\<\>\)"
+"" Basic
```

Comment:
Why has this been split out?

Reply:
It was easier to know what was going on this way, imo.
And the initial query was not correct since `%%=` would have been matched 3 times, highlighting it as a valid operator.